<launch>
  <arg name="which" default="rlev1obstacles" />
  <arg name="use_ackermann_rrt" default="1" />
  <arg name="use_ackermann_rrt2" default="0" />
  <arg name="replan" default="always" />
  <arg name="allow_reverse" default="0" />

  <node pkg="map_server" type="map_server" name="map_server" args="$(find lab5)/data/$(arg which).yaml" required="true" />

  <param name="robot_description" command="$(find xacro)/xacro '$(find racecar_description)/urdf/racecar.xacro'" />

  <node pkg="lab6" type="interactive_pose2d_tf.py" name="interactive_pose2d_tf">
    <remap from="~color" to="/robot_marker_color" />
  </node>

  <node pkg="lab6" type="rrt.py" name="rrt_planner" output="screen">
    <remap from="~map" to="/map" />
    <remap from="~local_map" to="/local_map" />
    <param name="~use_ackermann_rrt" value="$(arg use_ackermann_rrt)" />
    <param name="~use_ackermann_rrt2" value="$(arg use_ackermann_rrt2)" />
    <param name="~replan" value="$(arg replan)" />
    <param name="~allow_reverse" value="$(arg allow_reverse)" />
  </node>
  <node pkg="lab6" type="fake_cone.py" name="cone_emitter" output="screen">
    <remap from="~global_map" to="/map" />
    <remap from="~local_map" to="/local_map" />
    <remap from="~scan" to="/scan" />
  </node>


  <node name="rviz" pkg="rviz" type="rviz" args="-d '$(find lab6)/rviz/rrt.rviz'" required="true" />
</launch>
